OpenRA.Mods.Common/Traits/EngineerRepair.cs

```diff
@@ -80,46 +95,54 @@ public void ResolveOrder(Actor self, Order order)
 				self.CancelActivity();
 
 			self.SetTargetLine(order.Target, Color.Yellow);
-			self.QueueActivity(new RepairBuilding(self, order.Target, info.EnterBehaviour, info.ValidStances));
+			self.QueueActivity(new RepairBuilding(self, order.Target, Info));
 		}
 
 		class EngineerRepairOrderTargeter : UnitOrderTargeter
 		{
-			public EngineerRepairOrderTargeter()
-				: base("EngineerRepair", 6, "goldwrench", false, true) { }
+			EngineerRepairInfo info;
+
+			public EngineerRepairOrderTargeter(EngineerRepairInfo info)
+				: base("EngineerRepair", 6, info.Cursor, true, true)
+			{
+				this.info = info;
+			}
 
 			public override bool CanTargetActor(Actor self, Actor target, TargetModifiers modifiers, ref string cursor)
 			{
-				if (!target.Info.HasTraitInfo<EngineerRepairableInfo>())
+				var engiRepairable = target.Info.TraitInfoOrDefault<EngineerRepairableInfo>();
+				if (engiRepairable == null)
+					return false;
+
+				if (info.Types.Any() && engiRepairable.Types.Any() && !engiRepairable.Types.Overlaps(info.Types))
```

Comment:
In all other cases we only check the base bitset value, and we prefer `.IsEmpty` over `.Any()` in bitsets.

So this line should be `if (!engineerRepairable.Type.IsEmpty && !engineerRepairable.Type(info.Types))`